coriolis/tests/api/v1/views/test_minion_pool_view.py

```diff
@@ -0,0 +1,125 @@
+    """Test suite for the Coriolis api v1 views."""
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_minion_pool(
+        self,
+        mock_format_opt,
+    ):
+        mock_minion_pool_dict = {
+            'minion_machines': [{
+                'connection_info': {
+                    'pkey': 'mock_key',
+                    'password': 'mock_key',
+                    'certificates': {'key': 'mock_key'}
+                },
+                'backup_writer_connection_info': {
+                    'connection_details': {
+                        'pkey': 'mock_key',
+                        'password': 'mock_key',
+                        'certificates': {'key': 'mock_key'}
+                    },
+                }
+            }],
+        }
+        expected_result = {
+            'minion_machines': [{
+                'connection_info': {
+                    'pkey': '***',
+                    'password': '***',
+                    'certificates': {'key': '***'}
+                },
+                'backup_writer_connection_info': {
+                    'connection_details': {
+                        'pkey': '***',
+                        'password': '***',
+                        'certificates': {'key': '***'}
+                    },
+                }
+            }],
+        }
+        mock_format_opt.return_value = mock_minion_pool_dict
+
+        endpoint = mock.sentinel.endpoint
+        keys = mock.sentinel.keys
+        result = view._format_minion_pool(endpoint, keys)
+
+        mock_format_opt.assert_called_once_with(endpoint, keys)
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_minion_pool_connection_none(
+        self,
+        mock_format_opt,
+    ):
+        mock_minion_pool_dict = {
+            'minion_machines': [{
+                'connection_info': None,
+                'backup_writer_connection_info': {
+                    'connection_details': None,
+                }
+            }],
+        }
+        expected_result = {
+            'minion_machines': [{
+                'connection_info': None,
+                'backup_writer_connection_info': {
+                    'connection_details': None,
+                }
+            }],
+        }
+        mock_format_opt.return_value = mock_minion_pool_dict
+
+        endpoint = mock.sentinel.endpoint
+        keys = mock.sentinel.keys
+        result = view._format_minion_pool(endpoint, keys)
+
+        mock_format_opt.assert_called_once_with(endpoint, keys)
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    @mock.patch.object(view_utils, 'format_opt')
+    def test_format_minion_pool_no_minion_machines(
+        self,
+        mock_format_opt,
+    ):
+        mock_minion_pool_dict = {
+            'minion_machines': [],
```

Comment:
This is not what I was asking for. Since the code checks whether `minion_machines` is inside the dict, you should test a case where this key is non-existent, instead of having an empty list. 
https://github.com/cloudbase/coriolis/blob/2138d7b0ae1dfdbfe44d84206c2fc5dc3c9d58b9/coriolis/api/v1/views/minion_pool_view.py#L20